GIT_WORKFLOW.md

```diff
@@ -1 +1,40 @@
-Write based on: https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow
\ No newline at end of file
+# Git Workflow
+
+The Economist Digital Solutions teams use git for version control following the [Forking  Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow).
+
+The advantage of the Forking Workflow is that contributions can be integrated without the need for everybody to push to a single central repository. Developers push to their own repositories, and only maintainer teams can push to the official repository. This allows  maintainers to accept commits from any developer without giving them write access to the official codebase. This approach scales easily and encourages sharing and collaboration on code bases across teams with minimal overhead required to manage merging and branching.
```

Comment:
Is there ever a need for CP / WP to have branch based builds ? Like tests running on their branches by CI ? Is a PR environment sufficient enough ? Shouldn't your personal github have you environment too ? 

Sorry just trying to understand, like on mobile the branches allows us to have an artifact for every branch that is testable. The develop is the latest release (v x.y.z) and merge to master pushes it to our staging environement which is the stores for beta release allowing the final UAT / QA sign-off for release ?

Reply:
On the CP, we've only used branch based builds to maintain multiple versions of an API. Generally developers create a branch on their fork to do their work and run their local env against that branch for testing, but all PRs are against the main repo's master, and once they are approved, they are merged and tests run against master. The changes go to Stage for manual QA, and then to Prod after QA approval. 
Code is continually being integrated, features can be more easily broken into small changes that are tested in the full environment. What is the benefit of having a testable artifact for every branch? Tagging released on the commit history keeps the record.

Reply:
The reason we have a testable artifact on mobile because generally there are times where we need UX feedback before we can call it done ? Animations / Some text changes. It may not be the same as the designs provided and hence it is important to have this. By building every bracnh we have unlimited storage for our artifact so it is good for us. 

My only concern with Forking is the developer forks / then creates a branch on his fork and then creates a PR to the master branch. Where we can easily protect the master branch in github, developer creates a branch and creates a PR directly to the main branch. It is the responsibility of the tech-lead or core-reviewer to merge the branch.

Also, it is the job of the team to have a session at the end of the sprint to get rid of stale branches to keep git healthy ?

Reply:
Via the settings in github, you can set different levels of strictness and protection. For instance on e.com is not possible to merge a PR that has not been approved. This can be bypassed only from Admin in extreme cases. At the same way, you can disable direct push against master. I think this should always be set.